diff --git a/dist/runtime/plugins/colors.js b/dist/runtime/plugins/colors.js
index b8208bfff036bc737b6918ea664ca1c1ee38a5e6..902ff6269bb14f822fcda5a9a6f82a77bf933c60 100644
--- a/dist/runtime/plugins/colors.js
+++ b/dist/runtime/plugins/colors.js
@@ -36,7 +36,6 @@ export default defineNuxtPlugin(() => {
   const headData = {
     style: [{
       innerHTML: () => root.value,
-      tagPriority: -2,
       id: "nuxt-ui-colors"
     }]
   };
