diff --git a/dist/runtime/vue/stubs.js b/dist/runtime/vue/stubs.js
index 3bfdce3951a6d7b509377c3fd08d1f111f16cabc..6596ef2c83dd2015a9205fb4f477e26fdb425bb1 100644
--- a/dist/runtime/vue/stubs.js
+++ b/dist/runtime/vue/stubs.js
@@ -57,7 +57,7 @@ const hooks = createHooks();
 export function useNuxtApp() {
   return {
     isHydrating: true,
-    payload: { serverRendered: false },
+    payload: { serverRendered: import.meta.env.SSR || false },
     hooks,
     hook: hooks.hook
   };
